db/version_edit_handler.cc

```diff
@@ -889,13 +893,21 @@ void DumpManifestHandler::CheckIterationResult(const log::Reader& reader,
     fprintf(stdout, "%s\n", s->ToString().c_str());
     return;
   }
+  assert(cf_to_cmp_names_);
   for (auto* cfd : *(version_set_->column_family_set_)) {
     fprintf(stdout,
             "--------------- Column family \"%s\"  (ID %" PRIu32
             ") --------------\n",
             cfd->GetName().c_str(), cfd->GetID());
     fprintf(stdout, "log number: %" PRIu64 "\n", cfd->GetLogNumber());
-    fprintf(stdout, "comparator: %s\n", cfd->user_comparator()->Name());
+    auto it = cf_to_cmp_names_->find(cfd->GetID());
+    if (it != cf_to_cmp_names_->end()) {
```

Comment:
This is not exactly the same test as in the Extract case, since the extract case depended on has_comparator_.  Does that matter or not for these purposes?

Reply:
Sorry missed this.
The check in `ExtractInfoFromVersionEdit` is performed while reading MANIFEST. On seeing a version edit that contains comparator name (has_comparator_), ldb compares the comparator name with the one specified via `options` with the name recorded in the MANIFEST. If they do not match, it may just be because the comparator is a custom comparator which `ldb` does not know. In order to be able to print its name later, we save the `cf_id->comparator` mapping.
The two lines you quoted here are executed after reading the MANIFEST. At this point, the column family's comparator name is either equal to `cfd->user_comparator()->Name()`, or saved in the map. There must have been a version edit that has comparator, due to how we create column family in `CreateColumnFamilyImpl` and `WriteCurrentStateToManifest`.